simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/HeartbeatScheduler.java

```diff
@@ -19,24 +23,45 @@
     private StationState stationState;
     private StationMessageSender stationMessageSender;
 
+    private AtomicInteger heartBeatInterval;
+    private AtomicLong timeOfLastHeartbeatSent;
+
     public HeartbeatScheduler(StationState stationState, StationMessageSender stationMessageSender) {
         this.stationState = stationState;
         this.stationMessageSender = stationMessageSender;
+        this.heartBeatInterval = new AtomicInteger();
+        this.timeOfLastHeartbeatSent = new AtomicLong(0);
     }
 
-    public void scheduleHeartbeat(int heartbeatInterval) {
+    public void updateHeartbeat(int heartbeatInterval) {
         LOGGER.debug("Scheduling heartbeat to {} sec.", heartbeatInterval);
-        heartbeatExecutor.scheduleAtFixedRate(this::sendHeartbeat, heartbeatInterval, heartbeatInterval, TimeUnit.SECONDS);
```

Comment:
this check should run every second, regardless of the value of heartbeat interval